plugins/spu2-x/src/Linux/Config.cpp

```diff
@@ -124,12 +126,13 @@ void ReadSettings()
 
 #if SDL_MAJOR_VERSION >= 2
     CfgReadStr(L"OUTPUT", L"Output_Module", temp, SDLOut->GetIdent());
-#else
+#elif defined(SPU2X_PORTAUDIO)
```

Comment:
I think this change is incorrect. SDL1.2, no PortAudio SPU2-X builds will not work properly because the `OUTPUT/Output_Module` ini key won't be read.

Reply:
Hmm, yeah, sdl 1.2 is still a possibility, as onepad-legacy still supports it (as well as 2.0), even if onepad doesn't.

Reply:
How does someone test with SDL1?
```
CMake Error at cmake/SelectPcsx2Plugins.cmake:21 (message):
  Skip build of onepad: missing dependencies:check these libraries -> sdl2,
  X11, gtk2
```
It seems my previous test with SDL1 was mistaken and I had SDL2 installed.

Reply:
I got past that with `rm -r plugins/onepad` and `-DSDL2_API=FALSE`

However it now fails with many errors like this.
```
/tmp/SBo/pcsx2/build/plugins/onepad_legacy/ImgHeader/dualshock2.h:9:29: error: expected primary-expression before ';' token
  static const uint Length = ;
                             ^
/tmp/SBo/pcsx2/build/plugins/onepad_legacy/ImgHeader/dualshock2.h:10:29: error: size of array 'Data' is not an integral constant-expression
  static const u8 Data[Length];
                             ^
/tmp/SBo/pcsx2/build/plugins/onepad_legacy/ImgHeader/dualshock2.h:14:37: error: size of array 'Data' is not an integral constant-expression
 const u8 res_dualshock2::Data[Length] =
                                     ^
```
This is using `sdl-1.2.15` and I won't be able to confirm these concerns or try to fix them until this is fixed. It might be just better to remove SDL1 support instead? I'm not sure anyone does or should use it with pcsx2?

Reply:
Only real reason I can think of to leave SDL 1.2 support in there is for any platforms that have 1.2 but not 2.0, and while they did drop platforms in between the two, I'm pretty sure people shouldn't be trying to compile pcsx2 on, say, BeOS or an Amiga.

onepad does use both the 1.2 and 2.0 sdl api for controllers, but the 1.2 api is still in there, and isn't actually depreciated, just the 2.0 one is better for controllers as opposed to joysticks.

Either fixing onepad_legacy with sdl 1.2 or removing support is fairly outside the scope of this pr, though. As long as we allow for the possibility of it being sdl 1.2 in spu2-x for now, I think we can handle onepad_legacy being broken with sdl 1.2 as a separate issue later. :P

I'm not sure yet if I want to fix that, or just drop sdl 1.2 support.

Those are autogenerated files from images for all the pictures in the config screen, btw...

Reply:
Yea, I agree that its out of scope. I think the options are to leave the remaining SDL1 issues for later when they can be tested properly or to wait until they are fixed and then rebase this PR accordingly. I am not sure the latter option will ever be done unless someone who understands how these files are generated better can take the initiative.

Regardless I don't think using SDL1 with pcsx2 on Slackware is a good idea where SDL2 is readily available and this PR seems ready as far as SDL2 is concerned.

Reply:
The CMake error is poor handling on our end.

onepad_legacy compiles fine on my end (ArchLinux) with SDL1.2 without modifications. So the errors you had are a bit strange.

As for SDL1.2 support - some distros link WX to SDL1.2. Otherwise it probably would have been dropped already. The situation might have changed - the two distros that I know of (Fedora and OpenSUSE) that used to link WX to SDL1.2 now seem to link WX to SDL2 instead.

Reply:
It seems the real issue is was higher up in the log.
```
[ 34%] Generating ImgHeader/dualshock2.h
binmode() on closed filehandle IN at /tmp/SBo/pcsx2/linux_various/hex2h.pl line 63.
read() on closed filehandle IN at /tmp/SBo/pcsx2/linux_various/hex2h.pl line 64.
Use of uninitialized value $byte in numeric ne (!=) at /tmp/SBo/pcsx2/linux_various/hex2h.pl line 64.
Use of uninitialized value $filesize in concatenation (.) or string at /tmp/SBo/pcsx2/linux_various/hex2h.pl line 87.
```

It seems WX uses neither SDL1 or SDL2 here.

Reply:
@turtleli You were right, that change was not correct with SDL1 and without portaudio. I rebased against master and included a small adjustment in the first commit to fix this.